torch/csrc/api/src/nn/modules/transformer.cpp

```diff
@@ -223,5 +224,66 @@ Tensor TransformerDecoderLayerImpl::activation(const Tensor& input){
   }
 }
 
+
+// ========================TransformerEncoderImpl=========================
+TransformerEncoderImpl::TransformerEncoderImpl(
+  TransformerEncoderOptions options_) : options(std::move(options_)) {
+  reset();
+}
+
+void TransformerEncoderImpl::reset() {
+  layers = this->register_module("layers", ModuleList());
+  for (int64_t i = 0; i < options.num_layers(); ++i) {
+    layers->push_back(options.encoder_layer()->clone());
+  }
```

Comment:
I feel a bit weird about each layer having a copy of the same parameters, but it looks like that is the same thing as what we do in Python

Reply:
@zou3519 This is what transformer does. It followed the the method in a paper here:
https://papers.nips.cc/paper/7181-attention-is-all-you-need.pdf
I am not sure the reason behind. Maybe @zhangguanheng66 can explain here.

Reply:
TransformerEncoder is composed of multiple layers and each layer is independent from the others. During training, the learnable parameters on each layers will be trained differently. For the python one, we pass a transformer encoder layer to the constructor of the transformer encoder and make deep copy. We just initiate the layers with same parameters but this is not necessary.